packages/process/test.pony

```diff
@@ -0,0 +1,159 @@
+use "ponytest"
+use "files"
+
+actor Main is TestList  
+  new create(env: Env) => PonyTest(env, this)
+  new make() => None
+
+  fun tag tests(test: PonyTest) =>
+    test(_TestCat)
+    test(_TestStderr)
+    test(_TestWc)
+
+class iso _TestCat is UnitTest
+  fun name(): String =>
+    "writing to STDIN of a forked cat process"
+
+  fun apply(h: TestHelper) =>
+    let client = ProcessClient("one, two, three", h)
+    let notifier: ProcessNotify iso = consume client
+    let path: String = "/bin/cat"
+    
+    let args: Array[String] iso = recover Array[String](1) end
+    args.push("cat")
+    
+    let vars: Array[String] iso = recover Array[String](2) end
+    vars.push("HOME=/")
+    vars.push("PATH=/bin")
+    
+    let pm: ProcessMonitor = ProcessMonitor(consume notifier, path,
+      consume args, consume vars)
+    pm.write("one, two, three")
+    h.long_test(2_000_000_000)
+    pm.dispose()
+
+  fun timed_out(h: TestHelper) =>
+    h.complete(false)
+
+class iso _TestStderr is UnitTest
+  fun name(): String =>
+    "writing to STDERR of via a forked cat process that fails"
+
+  fun apply(h: TestHelper) =>
+    let client = ProcessClient("cat: foobaz: No such file or directory\n", h)
+    let notifier: ProcessNotify iso = consume client
+    let path: String = "/bin/cat"
+    
+    let args: Array[String] iso = recover Array[String](2) end
+    args.push("cat")
+    args.push("foobaz")
+    
+    let vars: Array[String] iso = recover Array[String](2) end
+    vars.push("HOME=/")
+    vars.push("PATH=/bin")
+    
+    let pm: ProcessMonitor = ProcessMonitor(consume notifier, path,
+      consume args, consume vars)
+    h.long_test(2_000_000_000)
+    pm.dispose()
+
+  fun timed_out(h: TestHelper) =>
+    h.complete(false)
+
+class iso _TestWc is UnitTest
+  fun name(): String =>
+    "capturing STDOUT and STDERR of forked wc process"
+
+  fun apply(h: TestHelper) =>
+    // we expect wc to give a count of
+    // 1 line, 9 words, 44 characters, filename foobar
+    let client = ProcessClient("       1       3      14\n", h) 
+
+    let path: String = "/usr/bin/wc"
+    
+    let args: Array[String] iso = recover Array[String](1) end
+    args.push("wc")
+    
+    let vars: Array[String] iso = recover Array[String](2) end
+    vars.push("HOME=/")
+    vars.push("PATH=/bin")
+    
+    let notifier: ProcessNotify iso = consume client
+    let pm: ProcessMonitor = ProcessMonitor(consume notifier, path,
+      consume args, consume vars)
+    pm.write("one two three\n")
+    h.long_test(2_000_000_000)
+    pm.dispose()
+
+  fun timed_out(h: TestHelper) =>
+    h.complete(false)
+
+class ProcessClient is ProcessNotify
```

Comment:
Since the tests are part of the package, helper classes like this should be made private (with a leading underscore) to prevent cluttering the namespace of the types provided by a package, and to preventusers of the package from trying to use them directly.